gprofiler/main.py

```diff
@@ -103,16 +107,28 @@ def _generate_output_files(
         collapsed_data: str,
         local_start_time: datetime.datetime,
         local_end_time: datetime.datetime,
-        flamegraph: bool,
     ) -> None:
         start_ts = get_iso8061_format_time(local_start_time)
         end_ts = get_iso8061_format_time(local_end_time)
         base_filename = os.path.join(self._output_dir, "profile_{}".format(end_ts))
+
         collapsed_path = base_filename + ".col"
         Path(collapsed_path).write_text(collapsed_data)
+
+        # point last_profile.col at the new file; and possibly, delete the previous one.
+        last_profile = os.path.join(self._output_dir, "last_profile.col")
+        prev_profile = Path(last_profile).resolve()
+        atomically_symlink(os.path.basename(collapsed_path), last_profile)
+        if self._rotating_output:
```

Comment:
```suggestion
        if self._rotating_output and prev_profile.name != Path(last_profile).name:
```
Otherwise if `self_rotating_output == True` it will just unlink the symlink forever.

Reply:
Since `last_profile.col` is a symlink, `Path(last_profile).resolve()` is never `last_profile.name` itself, but whatever the symlinks points to (which is not the symlinks itself).

Reply:
Or, I didn't understand what you mean here?

Reply:
Okay I see. On the first run the symlink is missing then `resolve()` returns the same file. Fixed

Reply:
Exactly